back-end/Database/Infrastructure/AsyncTenantedDocumentSession.cs

```diff
@@ -0,0 +1,179 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Raven.Client.Documents;
+using Raven.Client.Documents.Indexes;
+using Raven.Client.Documents.Linq;
+using Raven.Client.Documents.Session;
+using Raven.Client.Documents.Session.Loaders;
+using Raven.Yabt.Database.Models;
+
+namespace Raven.Yabt.Database.Infrastructure
+{
+	public class AsyncTenantedDocumentSession : IAsyncTenantedDocumentSession
```

Comment:
That is a lot of code (and ongoing maintenance every time you want to use a different aspect of RavendB)